### PR TITLE
feat: add ability to support additional gateway/domain

### PR DESCRIPTION
**Pull request by joelmccoy**
## Description

Adds ability to override and add an additional gateway to the nginx config.

## Related Issue

Relates to #98 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-k3d/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: docs/DNS.md [hunk 1]
```diff
@@ -28,3 +28,14 @@ uds run --set COREDNS_OVERRIDES="$COREDNS_OVERRIDES"
 ## Nginx Configuration
 
 Additionally, the package includes Nginx configuration that assumes the use of `uds.dev` as the base domain. This configuration is tailored to support the development environment setup, ensuring that Nginx correctly handles requests and routes them within the cluster, based on the `uds.dev` domain.
+
+### Nginx Additional Gateway Support
+
+You can add an additional Gateway to the Nginx configuration by overriding the helm value `customGateway.domainName` with the desired domain name.
+
+i.e. to add a custom gateway with the domain `uds2.dev` you would add the following toyour helm overrides for the `uds-dev-stack` chart:
```
Comment (catsby) [inline docs/DNS.md:36]: nit: if I'm not mistaken `i.e.` _typically(?)_ means "that is" or "in other words", and I think "For example" (or `E.g.`) works better here as we're giving an actual example and not simply restating the prior sentence in another way. 

```suggestion
For example, to add a custom gateway with the domain `uds2.dev` you would add the following helm overrides for the `uds-dev-stack` chart:
```

**Review by catsby — DISMISSED** [review]
👍 is this something we have or should have tests to verify the behavior of?

File: docs/DNS.md (hunk 1 above)
Comment (mjnagel) [reply docs/DNS.md]: I definitely abuse this and read as `in example` often - good callout 😅

File: chart/templates/nginx.yaml [hunk 2]
```diff
@@ -24,6 +27,12 @@ data:
             server ###ZARF_VAR_BASE_IP###.201:443; 
         }
 
+        {{- if .Values.customGateway | default false}}
+        upstream backend_custom_gateway {
+            server ###ZARF_VAR_BASE_IP###.202:443;
```
Comment (mjnagel) [inline chart/templates/nginx.yaml:32]: I think this is a fair assumption, but I believe this assumes that any custom gateway is deployed after admin/tenant (based on the IP allocation). Should be fine, just wanted to note it.

**Comment by mjnagel** [comment]
> > 👍 is this something we have or should have tests to verify the behavior of?
> 
> I'm not sure if we want to explicitly test this based on this comment: [#98 (comment)](https://github.com/defenseunicorns/uds-k3d/issues/98#issuecomment-2392455479)
> 
> but will defer to @mjnagel if anything extra is required here. Happy to clarify in the docs that this isn't tested as well if that helps at all.

@catsby / @joelmccoy definitely feel weird advocating for not testing this, but I think that is what I said in the comment 😆. I may have poorly stated it there - I think if there were an easy way to test this in CI I'd be good with adding testing, but made an assumption that the complexity around a new domain/cert might be non-trivial. I mostly don't want us to invest too much time in a feature that is not required by uds-core and uds packages in general - I see that as the primary place to focus uds-k3d development efforts and testing.

Also worth calling out, as-is I don't think we even have tests of the primary domain nginx pattern (uds.dev). We probably should track the need for a test on the nginx config in general as a future issue. If developing that test naturally made it easy to add a test of the custom domain pattern no issue also handling that.

**Comment by catsby** [comment]
> assumption that the complexity around a new domain/cert might be non-trivial

@mjnagel I was hoping that helm had some kind of easy way to check that "given a template and a config, does helm execute to create the result we expect" kind of thing, so testing the generated helm result and not so much the final product e.g. (😉) "does it respond to multiple domains". If we don't have that, or it's not practical, so be it 👍

**Comment by mjnagel** [comment]
@catsby ah - makes sense, yeah more of unit test type thing rather than an e2e validation. Definitely could add some helm testing with helm template or other tooling to validate.

**Review by catsby — APPROVED** [review]

**Pull request merged**